common/djangoapps/enrollment/views.py

```diff
@@ -98,24 +167,71 @@ def get(self, request, course_id=None):
 
 
 class EnrollmentListView(APIView):
-    """ Enrollment API List View for viewing all course enrollments for a user. """
+    """ 
+        **Use Cases**
 
-    authentication_classes = OAuth2Authentication, SessionAuthenticationAllowInactiveUser
-    permission_classes = permissions.IsAuthenticated,
-    throttle_classes = EnrollmentUserThrottle,
+            1. Get a list of all course enrollments for the currently logged in user.
 
-    def get(self, request):
-        """List out all the enrollments for the current user
+            2. Enroll the currently logged in user in a course.
 
-        Returns a JSON response with all the course enrollments for the current user.
+               Currently you can use this command only to enroll the user in "honor" mode.
 
-        Args:
-            request (Request): The GET request for course enrollment listings.
-            user (str): Get all enrollments for the specified user's username.
+               If honor mode is not supported for the course, the request fails and returns the available modes.
+
+        **Example Requests**:
 
-        Returns:
-            A JSON serialized representation of the user's course enrollments.
+            GET /api/enrollment/v1/enrollment
 
+            POST /api/enrollment/v1/enrollment{"course_details":{"course_id": "edX/DemoX/Demo_Course"}}
+
+        **Post Parameters**
+ 
+            * user:  The user ID of the currently logged in user. Optional. You cannot use the command to enroll a different user.
+
+            * course details: A collection that contains:
```

Comment:
The collection just contains the one item (course_id)? Just want to confirm nothing is missing here.